StardewArchipelago/Locations/CodeInjections/Modded/SVE/SVECutsceneInjections.cs

```diff
@@ -98,35 +108,39 @@ public static bool CheckForAction_LanceChest_Prefix(Chest __instance, Farmer who
         }
 
         // public void endBehaviors(string[] split, GameLocation location)
-        public static bool EndBehaviors_AddRailroadBoulderIfIridiumBomb_Prefix(string[] split, Event __instance)
+        public static bool EndBehaviors_AddSpecialOrderAfterEvent_Prefix(string[] split, Event __instance)
         {
             try
             {
-                if (__instance.id != RAILROAD_BOULDER_ID || !Game1.player.mailReceived.Contains("RailroadBoulderRemoved"))
+                if (!sveEventSpecialOrders.ContainsKey(__instance.id))
                 {
                     return true; // run original logic
                 }
                 //Change the key so it doesn't get deleted
-                var railroadBoulderOrder = SpecialOrder.GetSpecialOrder("Clint2", null);
-                railroadBoulderOrder.questKey.Value = RAILROAD_KEY;
-                Game1.player.team.specialOrders.Add(railroadBoulderOrder);
+                var specialOrder = SpecialOrder.GetSpecialOrder(sveEventSpecialOrders[__instance.id], null);
+                Game1.player.team.specialOrders.Add(specialOrder);
 
                 return true; // run original logic
             }
             catch (Exception ex)
             {
-                _monitor.Log($"Failed in {nameof(EndBehaviors_AddRailroadBoulderIfIridiumBomb_Prefix)}:\n{ex}", LogLevel.Error);
+                _monitor.Log($"Failed in {nameof(EndBehaviors_AddSpecialOrderAfterEvent_Prefix)}:\n{ex}", LogLevel.Error);
                 return true; // run original logic
             }
         }
 
         // Railroad Boulder Special Order won't load if Iridium Bomb is sent early, so we duplicate it so the player gets it.
-        private static void AppendFakeClintBoulderSpecialOrderKey()
+        public static bool UpdateSpecialOrders_StopDeletingSpecialOrders_Prefix()
```

Comment:
Can you add the function definition as a comment above this patch? Like the others?